cmd/osbuild-worker/main.go

```diff
@@ -120,18 +120,19 @@ func RunJob(token uuid.UUID, manifest distro.Manifest, targets []*target.Target,
 				continue
 			}
 
-			if options.Key == "" {
-				options.Key = token.String()
+			key := options.Key
+			if key == "" {
+				key = uuid.New().String()
```

Comment:
We should move this into weldr and make the key mandatory in the worker. In the future we'll do the registering in composer and then we need to know the key. Can be fixed in a follow-up.